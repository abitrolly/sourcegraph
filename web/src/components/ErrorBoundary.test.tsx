import React from 'react'
import { ErrorBoundary } from './ErrorBoundary'
import { mount } from 'enzyme'

jest.mock('mdi-react/ErrorIcon', () => 'ErrorIcon')
jest.mock('mdi-react/ReloadIcon', () => 'ReloadIcon')

const ThrowError: React.FunctionComponent = () => {
    throw new Error('x')
}

/** Throws an error that resembles the Webpack error when chunk loading fails.  */
const ThrowChunkError: React.FunctionComponent = () => {
    throw new Error('Loading chunk 123 failed.')
}

describe('ErrorBoundary', () => {
    test('passes through if non-error', () =>
        expect(
            mount(
                <ErrorBoundary location={null}>
                    <ThrowError />
                </ErrorBoundary>
            )
        ).toMatchSnapshot())

    test('renders error page if error', () =>
        expect(
            mount(
                <ErrorBoundary location={null}>
                    <span>hello</span>
                </ErrorBoundary>
            )
        ).toMatchSnapshot())

    test('renders reload page if chunk error', () =>
        expect(
            mount(
                <ErrorBoundary location={null}>
                    <ThrowChunkError />
                </ErrorBoundary>
            )
        ).toMatchSnapshot())
})
