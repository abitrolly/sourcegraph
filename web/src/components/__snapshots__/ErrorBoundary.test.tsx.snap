// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary passes through if non-error 1`] = `
<ErrorBoundary
  location={null}
>
  <HeroPage
    icon="ErrorIcon"
    subtitle={
      <div
        className="container"
      >
        <p>
          Sourcegraph encountered an unexpected error. If reloading the page doesn't fix it, contact your site admin or Sourcegraph support.
        </p>
        <p>
          <code
            className="text-wrap"
          >
            x
          </code>
        </p>
      </div>
    }
    title="Error"
  >
    <div
      className="hero-page "
    >
      <div
        className="hero-page__icon"
      >
        <ErrorIcon />
      </div>
      <div
        className="hero-page__title"
      >
        Error
      </div>
      <div
        className="hero-page__subtitle"
      >
        <div
          className="container"
        >
          <p>
            Sourcegraph encountered an unexpected error. If reloading the page doesn't fix it, contact your site admin or Sourcegraph support.
          </p>
          <p>
            <code
              className="text-wrap"
            >
              x
            </code>
          </p>
        </div>
      </div>
    </div>
  </HeroPage>
</ErrorBoundary>
`;

exports[`ErrorBoundary renders error page if error 1`] = `
<ErrorBoundary
  location={null}
>
  <span>
    hello
  </span>
</ErrorBoundary>
`;

exports[`ErrorBoundary renders reload page if chunk error 1`] = `
<ErrorBoundary
  location={null}
>
  <HeroPage
    icon="ErrorIcon"
    subtitle={
      <div
        className="container"
      >
        <p>
          Sourcegraph encountered an unexpected error. If reloading the page doesn't fix it, contact your site admin or Sourcegraph support.
        </p>
        <p>
          <code
            className="text-wrap"
          >
            Loading chunk 123 failed.
          </code>
        </p>
      </div>
    }
    title="Error"
  >
    <div
      className="hero-page "
    >
      <div
        className="hero-page__icon"
      >
        <ErrorIcon />
      </div>
      <div
        className="hero-page__title"
      >
        Error
      </div>
      <div
        className="hero-page__subtitle"
      >
        <div
          className="container"
        >
          <p>
            Sourcegraph encountered an unexpected error. If reloading the page doesn't fix it, contact your site admin or Sourcegraph support.
          </p>
          <p>
            <code
              className="text-wrap"
            >
              Loading chunk 123 failed.
            </code>
          </p>
        </div>
      </div>
    </div>
  </HeroPage>
</ErrorBoundary>
`;
